src/project/view/scorepropertiesmodel.cpp

```diff
@@ -0,0 +1,220 @@
+        if (!value.canConvert<QString>()) {
+            return false;
+        }
+        tag.name = value.toString();
+        return true;
+    case TagValue:
+        if (!value.canConvert<QString>()) {
+            return false;
+        }
+        tag.value = value.toString();
+        return true;
+    default:
+        return false;
+    }
+}
+
+int ScorePropertiesModel::rowCount(const QModelIndex&) const
+{
+    return m_tags.size();
+}
+
+QHash<int, QByteArray> ScorePropertiesModel::roleNames() const
+{
+    static const QHash<int, QByteArray> roles {
+        { TagName, "tagName" },
+        { TagValue, "tagValue" },
+        { IsStandardTag, "isStandardTag" }
+    };
+
+    return roles;
+}
+
+QString ScorePropertiesModel::filePath() const
+{
+    return m_projectMetaInfo.filePath.toQString();
+}
+
+QString ScorePropertiesModel::version() const
+{
+    return m_projectMetaInfo.musescoreVersion;
+}
+
+QString ScorePropertiesModel::revision() const
+{
+    int rev = m_projectMetaInfo.musescoreRevision;
+    QString revision;
+    if (rev > 99999) {     // MuseScore 1.3 is decimal 5702, 2.0 and later uses a 7-digit hex SHA
+        revision = QString::number(rev, 16);
+    } else {
+        revision = QString::number(rev, 10);
+    }
+
+    return revision;
+}
+
+QString ScorePropertiesModel::apiLevel() const
+{
+    return QString::number(m_projectMetaInfo.mscVersion);
+}
+
+void ScorePropertiesModel::newTag()
+{
+    int destinationIndex = m_tags.size();
+    beginInsertRows(QModelIndex(), destinationIndex, destinationIndex);
+
+    Tag tag = { "Property", "Value", false };
+    m_tags.append(tag);
+
+    endInsertRows();
+
+    emit tagAdded(destinationIndex);
+}
+
+void ScorePropertiesModel::deleteTag(int index)
+{
+    if (index < 0 && index >= m_tags.size()) {
+        return;
+    }
+
+    beginRemoveRows(QModelIndex(), index, index);
+
+    m_tags.removeAt(index);
+
+    endRemoveRows();
+}
+
+void ScorePropertiesModel::saveTags()
+{
+    INotationProjectPtr project = context()->currentProject();
+    if (!project) {
+        return;
+    }
+
+    ProjectMeta meta;
+
+    meta.title = m_tags[0].value;
+    meta.arranger = m_tags[1].value;
+    meta.composer = m_tags[2].value;
+    meta.copyright = m_tags[3].value;
+    meta.creationDate = QDate::fromString(m_tags[4].value);
+    meta.lyricist = m_tags[5].value;
+    meta.translator = m_tags[6].value;
+    meta.platform = m_tags[7].value;
+    meta.source = m_tags[8].value;
+
+    for (Tag tag : m_tags) {
+        if (tag.isStandardTag) {
+            continue;
+        }
+
+        meta.additionalTags[tag.name] = tag.value;
+    }
+
+    project->setMetaInfo(meta, true);
+}
+
+void ScorePropertiesModel::openFileLocation()
+{
+    std::string path = m_projectMetaInfo.filePath.toStdString();
+    int position = int(path.find(m_projectMetaInfo.fileName().toStdString()));
+
+    Ret ret = interactive()->openUrl(m_projectMetaInfo.filePath.toStdString().substr(0, position));
```

Comment:
Could you please explain, what is the purpose of these manipulations with the file path?

Reply:
I think you simply want to use `io::absoluteDirpath(m_projectMetaInfo.filePath)`.

Reply:
> Could you please explain, what is the purpose of these manipulations with the file path?

It was to get the path of the parent folder. The code "slices" off the file name part of the url. So `score.mscz` would be cut off from `C://Scores/score.mscz`. But, @cbjeukendrup's suggestion is better. So I will use that